src/core/src/pass/serialize.cpp

```diff
@@ -131,7 +115,7 @@ class ConstantWriter {
             // the same hash for {2, 2} and {0, 128} arrays.
             // But even strong hashing algorithms sometimes give collisions.
             // Therefore we always have to compare values when finding a match in the hash multimap.
-            const HashValue hash = hash_combine(ptr_to_write, *new_size);
+            const HashValue hash = ov::runtime::combine_hash(ptr_to_write, *new_size);
```

Comment:
1. does this function provide the same results as original `hash_combine` ?
2. If yes, could you please add such tests? It's important to preserve the same results for hashes to avoid regeneration of compiled_blobs on customers side with newer OpenVINO release.

Reply:
If this jit implement for hash computation is good enough, can we also extend to other component's hash computation? As we know, there are many duplicated similar combine_hash functions, such as in `compilation_context.cpp`, `ov::util::hash_combine`, `ov::snippets::pass::<unnamed>::hash_combine()`, can we merge them into a single version?

Reply:
Ilya,
1. No, that is another approach,
2. Cache is recompiled on each commit change, due to `openvino_version` and `runtime_info` contain commit hash,
3. Please take a look at file `Compile_model_time_cache_HASH_FIX_branch_vs_master.xlsx` in the ticket 127331.

Reply:
River,
It's good point for further optimizations.